vm/src/stdlib/unicodedata.rs

```diff
@@ -0,0 +1,94 @@
+/* Access to the unicode database.
+   See also: https://docs.python.org/3/library/unicodedata.html
+*/
+
+use crate::function::OptionalArg;
+use crate::obj::objstr::PyStringRef;
+use crate::pyobject::{PyObjectRef, PyResult};
+use crate::vm::VirtualMachine;
+
+use unic::char::property::EnumeratedCharProperty;
+use unic::ucd::category::GeneralCategory;
+use unic::ucd::Name;
+use unicode_names2;
+
+pub fn make_module(vm: &VirtualMachine) -> PyObjectRef {
+    let ctx = &vm.ctx;
+
+    let unidata_version = unic::UNICODE_VERSION.to_string();
+
+    py_module!(vm, "unicodedata", {
+        "bidirectional" => ctx.new_rustfunc(bidirectional),
+        "category" => ctx.new_rustfunc(category),
+        "name" => ctx.new_rustfunc(name),
+        "lookup" => ctx.new_rustfunc(lookup),
+        "normalize" => ctx.new_rustfunc(normalize),
+        "unidata_version" => ctx.new_str(unidata_version),
+    })
+}
+
+fn category(character: PyStringRef, vm: &VirtualMachine) -> PyResult {
+    let my_char = extract_char(character, vm)?;
+    let category = GeneralCategory::of(my_char);
+    Ok(vm.new_str(category.abbr_name().to_string()))
+}
+
+fn lookup(name: PyStringRef, vm: &VirtualMachine) -> PyResult {
+    // TODO: we might want to use unic_ucd instead of unicode_names2 for this too, if possible:
+    if let Some(character) = unicode_names2::character(&name.value) {
+        Ok(vm.new_str(character.to_string()))
+    } else {
+        Err(vm.new_key_error(format!("undefined character name '{}'", name)))
+    }
+}
+
+fn name(
+    character: PyStringRef,
+    default: OptionalArg<PyObjectRef>,
+    vm: &VirtualMachine,
+) -> PyResult {
+    let my_char = extract_char(character, vm)?;
+
+    if let Some(name) = Name::of(my_char) {
+        Ok(vm.new_str(name.to_string()))
+    } else {
+        match default {
```

Comment:
You can use `unwrap_or` here to make it nicer.

Reply:
Okay, but I'll keep it this way, its probably clear enough.